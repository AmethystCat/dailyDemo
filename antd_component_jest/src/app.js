import React from 'react';
import { Select } from 'antd';

const { Option } = Select;

export class T extends React.Component {
  render() {
    return <Select style={{width: 200}} defaultValue={"0"}>
      <Option value="0">Lily</Option>
      <Option value="1">Lucy</Option>
    </Select>
  }
}