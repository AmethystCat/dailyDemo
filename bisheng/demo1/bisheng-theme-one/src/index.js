const path = require('path');
console.log(123);
module.exports = {
  lazyLoad: true,
  pick: {
    posts(markdownData) {
      console.log('bisheng-theme-one/src/index', markdownData);
      return {
        meta: markdownData.meta,
        description: markdownData.description,
      };
    },
  },
  plugins: [
    path.join(__dirname, '../..', 'node_modules', 'bisheng-plugin-description'),
    path.join(__dirname, '../..', 'node_modules', 'bisheng-plugin-react?lang=jsx'),
  ],
  routes: [{
    path: '/',
    component: './template/Archive',
  }, {
    path: '/posts/:post',
    dataPath: '/:post',
    component: './template/Post',
  },{
    path: '/posts/:post/:sub',
    dataPath: '/:post/:sub',
    component: './template/Post',
  },{
    path: '/tags',
    component: './template/TagCloud',
  }],
};
