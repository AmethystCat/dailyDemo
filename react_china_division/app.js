import React from 'react';
import ReactDOM from 'react-dom';
import moment from 'moment';
import { Cascader, version } from 'antd';
import 'antd/dist/antd.css';

import provinces from 'china-division/dist/provinces.json';
import cities from 'china-division/dist/cities.json';
import areas from 'china-division/dist/areas.json';

console.log(provinces, cities, areas);

areas.forEach(area => {
  const matchCity = cities.filter(city => city.code === area.cityCode)[0];
  if (matchCity) {
    matchCity.children = matchCity.children || [];
    matchCity.children.push({
      label: area.name,
      value: area.code
    });
  }
});

cities.forEach(city => {
  const matchProvince = provinces.filter(
    province => province.code === city.provinceCode
  )[0];
  if (matchProvince) {
    matchProvince.children = matchProvince.children || [];
    matchProvince.children.push({
      label: city.name,
      value: city.code,
      children: city.children
    });
  }
});

const options = provinces.map(province => ({
  label: province.name,
  value: province.code,
  children: province.children
}));

console.log(options);

const onChangeHandler = (...args) => {
  console.log(args);
  // getLocation();
};

ReactDOM.render(
  <div style={{ margin: 24 }}>
    <p style={{ marginBottom: 24 }}>
      Current antd version: {version} <br />
      You can change antd version on the left panel (Dependencies section).
    </p>
    <Cascader
      options={options}
      showSearch
      placeholder="省 / 市 / 区"
      style={{ width: 400 }}
      onChange={onChangeHandler}
    />
  </div>,
  document.getElementById('app')
);

function getLocation(location) {
  // 地址解析
  // window.AMap.service('AMap.Geocoder', function() {
  //   //回调函数
  //   console.log(AMap);
  //   console.log(AMap.Geocoder);
    //实例化Geocoder
    // var geocoder = new AMap.Geocoder({
    //   city: '010' //城市，默认：“全国”
    // });
    // geocoder.getLocation('四川省成都市高新区天府软件园E区', (status, result) => {
    //   console.log(status, result);
    // });
    //TODO: 使用geocoder 对象完成相关功能
  // });
}
