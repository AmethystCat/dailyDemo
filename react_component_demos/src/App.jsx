import React from 'react';
import Menu from './menu-layout';
import Img from './img-view';
import Scroll from './scroll';
import { BrowserRouter as Router, Switch, Route, Link } from 'react-router-dom';

const data = [
  { id: '1', url: 'http://img0.imgtn.bdimg.com/it/u=2923679630,1264534690&fm=26&gp=0.jpg' },
  { id: '2', url: 'http://img5.imgtn.bdimg.com/it/u=1455747444,2513081477&fm=26&gp=0.jpg' },
  { id: '3', url: 'http://img0.imgtn.bdimg.com/it/u=1300810919,2064879991&fm=26&gp=0.jpg' },
  { id: '4', url: 'http://img4.imgtn.bdimg.com/it/u=3450230800,3260541970&fm=26&gp=0.jpg' },
  { id: '5', url: 'http://img5.imgtn.bdimg.com/it/u=3796740409,1068808904&fm=26&gp=0.jpg' },
  { id: '6', url: 'http://img5.imgtn.bdimg.com/it/u=2893150818,3423535185&fm=26&gp=0.jpg' },
  { id: '7', url: 'http://img5.imgtn.bdimg.com/it/u=1220092432,2680710673&fm=26&gp=0.jpg' },
  { id: '8', url: 'http://img1.imgtn.bdimg.com/it/u=330109828,109858191&fm=26&gp=0.jpg' },
  { id: '9', url: 'http://img2.imgtn.bdimg.com/it/u=1327495826,1844051418&fm=26&gp=0.jpg' },
  { id: '10', url: 'http://img3.imgtn.bdimg.com/it/u=3027296133,1203448824&fm=26&gp=0.jpg' }
];

const App = () => {
  return (
    <div className="container">
      <Router>
        <nav>
          <ul>
            <li>
              <Link to="/menu">menu</Link>
            </li>
            <li>
              <Link to="/img">img</Link>
            </li>
            <li>
              <Link to="/scroll">scroll</Link>
            </li>
          </ul>
        </nav>

        {/* A <Switch> looks through its children <Route>s and
            renders the first one that matches the current URL. */}
        <Switch>
          <Route path="/menu">
            <Menu />
          </Route>
          <Route path="/img">
            <Img data={data} />
          </Route>
          <Route path="/scroll">
            <Scroll />
          </Route>
        </Switch>
      </Router>
    </div>
  );
};

export default App;
