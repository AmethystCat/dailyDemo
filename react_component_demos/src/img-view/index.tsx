import React, { FC, useState } from 'react';
import { Carousel } from 'antd-mobile';
import 'antd-mobile/dist/antd-mobile.css';
import './img.css';

interface IImg {
  id: string;
  url: string;
}
interface IImgViewProps {
  data: IImg[];
  MAX?: 3 | 6 | 9; // 最大支持展示的数量，默认为九宫格
}

const screenRatio = window.innerHeight / window.innerWidth;
const ImgView: FC<IImgViewProps> = ({ data, MAX = 9 }) => {
  const [carousel, setCarousel] = useState({ visible: false, activeIndex: 0 });
  const remain = data.length - MAX;
  const imgs = remain > 0 ? data.slice(0, MAX) : data;

  return (
    <>
      <div className="section-img">
        {imgs.map((img, index) => {
          return (
            <div key={img.id} className="img" onClick={() => setCarousel({ visible: true, activeIndex: index })}>
              {remain > 0 && index === MAX - 1 && <div className="mask">+{remain}</div>}
              <img src={img.url} alt="" />
            </div>
          );
        })}
      </div>
      {carousel.visible && (
        <div className="section-carousel" onClick={() => setCarousel({ visible: false, activeIndex: 0 })}>
          <Carousel autoplay={false} infinite selectedIndex={carousel.activeIndex}>
            {data.map(img => (
              // 由于carousel会根据子元素的高度动态设置轮播元素的容器高度，
              // 这里统一将图片的外层div.img-wrapper固定设为屏幕高度，这样可以使轮播容器撑开，然后通过flex使图片垂直居中
              <div className="img-wrapper" key={img.id} style={{ height: window.innerHeight, width: '100%' }}>
                <img
                  src={img.url}
                  id={'img' + img.id}
                  onLoad={() => {
                    let imgEl = document.querySelector('#img' + img.id) as HTMLImageElement;
                    let { naturalHeight: nH, naturalWidth: nW } = imgEl;
                    let imgRatio = nH / nW;

                    // 如果图片的高宽比 大于 屏幕的高宽比，则将图片的高度(height)设为屏幕高度，宽度自适应
                    // 否则宽度设为屏幕的宽度(100%)，高度自适应
                    if (imgRatio > screenRatio) {
                      imgEl.style.height = window.innerHeight + 'px';
                      imgEl.style.width = 'auto';
                    }
                    // fire window resize event to change height
                    window.dispatchEvent(new Event('resize'));
                  }}
                />
              </div>
            ))}
          </Carousel>
        </div>
      )}
    </>
  );
};

export default ImgView;
