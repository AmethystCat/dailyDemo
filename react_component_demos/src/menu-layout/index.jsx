import React, { useState } from 'react';
import { menu, listObj } from './data';
import Menu from './menu';
import List from './list';
import './menu.css';

const defaultMenu = '素菜';
const MenuLayout = () => {
  const [activeMenu, setActiveMenu] = useState(defaultMenu);
  return (
    <div className="menu-w">
      <Menu
        data={menu}
        currMenu={activeMenu}
        onMenuClick={menuName => {
          let anchorElement = document.getElementById(menuName);
          // 如果对应id的锚点存在，就跳转到锚点
          if (anchorElement) {
            anchorElement.scrollIntoView({ block: 'start', behavior: 'smooth' });
          }
          setActiveMenu(menuName);
        }}
      />
      <List data={listObj} />
    </div>
  );
};

export default MenuLayout;
