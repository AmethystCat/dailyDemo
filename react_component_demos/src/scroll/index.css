.wrapper {
  height: 400px;
  border: 1px solid #ccc
}
#bs {
  height: 100%;
  padding: 0 10px;
  overflow: hidden;
}

.pullup-list {
  padding: 0;
}
.pullup-list-item {
  padding: 10px 0;
  list-style: none;
  border-bottom: 1px solid #ccc;
}

.pullup-tip-wrapper {
  padding: 10px;
  padding-top: 0;
  text-align: center;
  color: #999;
}
