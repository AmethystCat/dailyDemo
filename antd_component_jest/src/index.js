import React from 'react';
import ReactDOM from 'react-dom';
import { T } from './app';

const A = () => <div><h2>React</h2></div>;

ReactDOM.render(<T />, document.querySelector('#app'));