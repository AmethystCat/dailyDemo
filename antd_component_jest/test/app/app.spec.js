import React from 'react';
import TestRenderer from 'react-test-renderer';
import { T } from '../../src/app';

describe('app snapshot', () => {
  test('test', () => {
    const component = TestRenderer.create(<T />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});