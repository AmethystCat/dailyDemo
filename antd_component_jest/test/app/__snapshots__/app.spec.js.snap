// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app snapshot test 1`] = `
<div
  className="ant-select ant-select-enabled"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseDown={[Function]}
  onTouchStart={[Function]}
  style={
    Object {
      "width": 200,
    }
  }
>
  <div
    aria-autocomplete="list"
    aria-expanded={false}
    aria-haspopup="true"
    className="ant-select-selection
            ant-select-selection--single"
    onKeyDown={[Function]}
    role="combobox"
    tabIndex={0}
  >
    <div
      className="ant-select-selection__rendered"
    >
      <div
        className="ant-select-selection-selected-value"
        style={
          Object {
            "display": "block",
            "opacity": 1,
          }
        }
        title="Lily"
      >
        Lily
      </div>
    </div>
    <span
      className="ant-select-arrow"
      onClick={[Function]}
      style={
        Object {
          "WebkitUserSelect": "none",
          "userSelect": "none",
        }
      }
      unselectable="on"
    >
      <b />
    </span>
  </div>
</div>
`;
